lib/tessel/deploy.js

```diff
@@ -594,7 +604,7 @@ actions.writeToFile = function(t, entryPoint) {
       // If it was opened successfully
       .then(function(remoteProcess) {
         // When the remote process finishes
-        remoteProcess.stdin.once('finish', function() {
+        remoteProcess.once('close', function() {
```

Comment:
We should wait for the entire process to close to be sure the file was saved. Otherwise, we could see intermittent failures where we have finished writing data to stdin but the file has not yet been saved.